internal/cache/backend.go

```diff
@@ -43,14 +43,27 @@ func (b *Backend) Remove(ctx context.Context, h restic.Handle) error {
 	return b.Cache.Remove(h)
 }
 
-var autoCacheTypes = map[restic.FileType]struct{}{
-	restic.IndexFile:    {},
-	restic.SnapshotFile: {},
+func autoCacheTypes(ft restic.FileType, bt restic.BlobType) bool {
+	if ft == restic.IndexFile || ft == restic.SnapshotFile ||
+		(ft == restic.DataFile && bt != restic.DataBlob) {
+		return true
+	}
+	return false
 }
 
 // Save stores a new file in the backend and the cache.
+// Use SaveData when saving data files
 func (b *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindReader) error {
-	if _, ok := autoCacheTypes[h.Type]; !ok {
+	return b.save(ctx, h, restic.InvalidBlob, rd)
+}
+
+// Save stores a new data file in the backend and the cache.
+func (b *Backend) SaveData(ctx context.Context, h restic.Handle, tpe restic.BlobType, rd restic.RewindReader) error {
```

Comment:
I'm not convinced by that change. It requires differentiating between using `Save` and `SaveData`. I agree that the manual cache management in the `packer_manager` is not really optimal. But replacing a single use method with another, just makes the problem look different but won't solve it. Now we have the `Cache`, the `Backend` and a `CacheBackend`. 

The main problem with `PerformReadahead` in regards to the `packer_manager` is that it can't be updated. So my suggestion would be to modify the `Cache` and add a `AddReadahead(h restic.Handle)` method to it which replaces the `PerformReadahead` method. Then the `packer_manager` would just have to register the pack file before uploading and we're done.